src/updaters/expo/app-json.ts

```diff
@@ -0,0 +1,90 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {jsonStringify} from '../../util/json-stringify';
+import {logger as defaultLogger, Logger} from '../../util/logger';
+import {DefaultUpdater, UpdateOptions} from '../default';
+import {Version} from '../../version';
+
+export interface AppJson {
+  expo: {
+    version: string;
+    ios: {
+      buildNumber: string;
+    };
+    android: {
+      versionCode: string;
+    };
+  };
+}
+
+export interface AppJsonOptions extends UpdateOptions {
+  expoSDKVersion: Version;
+}
+
+/**
+ * This updates a React Natve Expo project app.json file's main, ios and android
+ * versions. All values except the `android.versionCode` are standard semver
+ * version numbers. For the `android.versionCode`, the semver number is used as
+ * the basis for the `versionCode`.
+ */
+export class AppJson extends DefaultUpdater {
+  expoSDKVersion: Version;
+  constructor(options: AppJsonOptions) {
+    super(options);
+    this.expoSDKVersion = options.expoSDKVersion;
+  }
+  /**
+   * Given initial file contents, return updated contents.
+   */
+  updateContent(content: string, logger: Logger = defaultLogger): string {
+    const parsed = JSON.parse(content) as AppJson;
+
+    logger.info(
+      `updating Expo version from ${parsed.expo.version} to ${this.version}`
+    );
+    parsed.expo.version = this.version.toString();
+
+    logger.info(
+      `updating iOS version from ${parsed.expo.ios.buildNumber} to ${this.version}`
+    );
+    parsed.expo.ios.buildNumber = this.version.toString();
```

Comment:
Question: does this also work if the app.json doesn't have this data (`ios.buildnumber`)? Expo doesn't require you to have it